### ISSUE TITLE
How to get localized date?

### ISSUE THREAD

**Issue by antonisprovidakis** [iamkun/dayjs#1587]
Hello, I've got a question.

Is there any way to get a date (year, month,day) in localized format without loading 
the desired locale first `require('dayjs/locale/de') etc.` ?

As in: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat#using_locales

```js
var date = new Date(Date.UTC(2012, 11, 20, 3, 0, 0));

// US English uses month-day-year order
console.log(new Intl.DateTimeFormat('en-US').format(date));
// → "12/19/2012"

// British English uses day-month-year order
console.log(new Intl.DateTimeFormat('en-GB').format(date));
// → "19/12/2012"
```

**Comment by mandaputtra** [comment]
Something like this, on the [docs](https://day.js.org/docs/en/timezone/timezone)

```
import dayjs from 'dayjs'
import utc from 'dayjs/plugin/utc'
import timezone from 'dayjs/plugin/timezone'

dayjs.extend(utc)
dayjs.extend(timezone)

dayjs.tz("2014-06-01 12:00", "America/New_York")
```

**Comment by antonisprovidakis** [comment]
No, that's not what I really need. What you propose is a way to transform a date to a specific timezone.

**Comment by z1haze** [comment]
I have the same issue. Even using `dayjs.tz.setDefault('America/New_York');` to set the timezone, and creating a new date still creates the date object with my local time.. and the docs regarding parsing a date in a particular timezone do not show anything about how to display an existing date in a specifc timezone.. at a loss here.

**Comment by mandaputtra** [comment]
@z1haze did you save the date with `tz` ?

try this?

```
dayjs(new Date().toISOString()).tz('America/New York')
```

**Comment by antonisprovidakis** [comment]
The issue is not about the timezone or local time.

It's about the different representations of the same date. Please the question.

Is there a `dayjs` way that I can have the same result as shown in the question, but without loading a specific locale (en-gb, el, etc.) first? 

I mean, I can do it with the native API:
```js
console.log(new Intl.DateTimeFormat('en-GB').format(date));
// → "19/12/2012"
```
But want to know if there is a `dayjs` way, without loading a locale first?
Otherwise, I have to mix native API and dayjs api.

**Comment by mandaputtra** [comment]
Oh, sorry I'm not getting it. 

No, you can't do that without the extend locale.

**Comment by MirzaLeka** [comment]
I think what @antonisprovidakis is referring to is reordering the date based on locale.
For example, in Moment there is a moment.locale() method that does just that. If you have something like

```js
import * as moment from 'moment'

function getDateLocale(locale) {
  moment.locale(locale);
  const date = new Date(2020, 4, 14);
  const momentDate = moment(date);
  return momentDate.format('L');
}

getDateLocale('en-GB'); //  14/05/2020
getDateLocale('en-US'); //  05/14/2020

```

See how the order changed?

This is what I tried, but it didn't help:

```js
import * as dayjs from 'dayjs'
import * as LocalizedFormat from 'dayjs/plugin/localizedFormat';

function getDateLocale(locale) {
  dayjs.extend(LocalizedFormat)
  dayjs.locale(locale)
  const date = new Date(2020, 4, 14);
  const dayJSDate = dayjs(date);
  return dayJSDate.format('L');
}

getDateLocale('en-GB'); //  05/14/2020
getDateLocale('en-US'); //  05/14/2020
```

How can we achieve the same in DayJS?

**Comment by antonisprovidakis** [comment]
I think that the part `import 'dayjs/locale/*';` is not what should be done here.

You have to import the a specific locale first:
```js
import 'dayjs/locale/en-gb'
```

These are all the available locales: https://github.com/iamkun/dayjs/tree/dev/src/locale

**Comment by MirzaLeka** [comment]
> You have to import a specific locale first:
> 
> ```js
> import 'dayjs/locale/en-gb'
> ```
> 
> These are all the available locales: https://github.com/iamkun/dayjs/tree/dev/src/locale

Yes, I figured when it crashed my app 😁
Anyway, I picked up another solution from [Deno support](https://github.com/iamkun/dayjs/issues/1382#issuecomment-792074819) Issue.
And also thanks to your reply, it works now:

```
import * as dayjs from 'dayjs'
import * as LocalizedFormat from 'dayjs/plugin/localizedFormat';
import 'dayjs/locale/en-gb';

function getDateLocale(locale) {
  dayjs.extend(LocalizedFormat)
  const date = new Date(2020, 4, 14);
  const dayJSDate = dayjs(date);
  return dayJSDate.locale(lang).format('L');
}

getDateLocale('en-GB'); //  14/05/2020
getDateLocale('en-US'); //  05/14/2020

```

But does this mean that we have to import every locale we intend to use?
My app supports 40+ languages 😢
[Full list of locales](https://unpkg.com/dayjs@1.10.6/locale.json)

**Comment by antonisprovidakis** [comment]
No, you can load a new locale by loading a module dynamically.

Something like this: https://webpack.js.org/api/module-methods/#dynamic-expressions-in-import.

I don't know how the final result will look like, as I haven't implemented this myself

**Comment by MirzaLeka** [comment]
> @antonisprovidakis did you find a solution to your problem? I am trying to get the local time based on time zone and language but it is not working
> 
> ```
> dayjs().locale('zh-cn').tz('Asia/Hong_Kong').format("LT")
> ```
> 
> Should print something like this : `02:10` but it prints `2:10 AM` - AM and PM is not used in Hong Kong
> 
> In moment this is possible: http://jsfiddle.net/j0pf89u6/
> 
> ```
> moment.locale('zh-cn');
> var now = moment().tz('Asia/Hong_Kong').format("LT")
> console.log(now);
> // 02:10
> ```

I couldn't figure out the way to remove it without doing something with JS.

However, what I decided to do on my side is keep the AM/PM and convert it to the Chinese equivalent with the use [localeData](https://day.js.org/docs/en/plugin/locale-data).
This LocaleData returns information about locales like names of weekdays. There is this helper function called meridiem that returns AM/PM that fits the right Locale.

Assuming you have the _zh-cn_ locale [imported](https://day.js.org/docs/en/installation/typescript#locale-and-plugin-import), you can do something like this

`dayjs( DateVariable ).locale('zh-cn').localeData().meridiem()` // 晚上
and then concat the output or replace the previous AM/PM  with this one.

The downside of this is that it's only available for handful of locales.
For everything that is not on the list below, calling meridiem will throw an error saying that is not a function. So you may want to [update ](https://day.js.org/docs/en/plugin/update-locale#docsNav) certain Locales and add custom meridiem logic.
![image](https://user-images.githubusercontent.com/23176181/138063401-efff39fb-c93f-43fd-82ee-c432dca83732.png)

**Comment by roshan-safertek** [comment]
have anyone got the answer @antonisprovidakis

**Comment by cfacundo** [comment]
in case this helps anyone, this shorter version of @MirzaLeka worked for me without the need of importing each `locale`:

```js
  import dayjs from 'dayjs'
  import LocalizedFormat from 'dayjs/plugin/localizedFormat'

  function getDateLocale(date: string) {
    dayjs.extend(LocalizedFormat)
    const dayJSDate = dayjs(date)
    return dayJSDate.locale(lang.value || 'es').format('L')
  }

```
